tgui/packages/tgui/interfaces/Orbit.js

```diff
@@ -82,14 +89,14 @@ export const Orbit = (props, context) => {
   const {
     alive,
     antagonists,
-    auto_observe,
     dead,
     ghosts,
     misc,
     npcs,
   } = data;
 
   const [searchText, setSearchText] = useLocalState(context, "searchText", "");
+  const [autoObs, setAutoObs] = useLocalState(context, "autoObs", false);
```

Comment:
You really should pass the auto observe data through props, I don't know what stylemistake thinks but I think the useLocalState with matching keys thing is super confusing.

Reply:
It's ok to use the same key. If it is repetitive, it should be refactored into a separate function like so:

```ts
const useAutoObserve = (context: any) => {
  return useLocalState(context, 'autoObserve, false);
};
```

Which then can be used as:

```ts
const [autoObserve, setAutoObserve] = useAutoObserve(context);
```

And for the love of god, don't abbreviate variable names like this, it's cringe.

Reply:
Also yes, this variable should really belong to ui_data, because you're passing it via act back to the server anyway.

Reply:
Give me more than an 80 character width to work with then. This is what you have wrought with your code golfing ways. Now I've got to arbitrarily line break them. How sad.

Reply:
```ts
const [autoObserve, setAutoObserve]
  = useLocalState(context, "autoObserve", false);
```

Reply:
Changed to pass via props. Expanded name to autoObserve. Praying linter doesn't whine as I went over the 80 char line limit to do it.

Reply:
> Changed to pass via props.

... ok